bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java

```diff
@@ -212,10 +214,13 @@ protected void stop() {
         if (started) {
             discoverComponents.stopDiscovery();
             delayedProcessing.join();
+            @NonNull
+            Collector<@NonNull CompletableFuture<@Nullable Void>, @NonNull Set<@NonNull CompletableFuture<@Nullable Void>>, @NonNull CompletableFuture<@Nullable Void>> allOfCollector = FutureCollector
+                    .allOf();
```

Comment:
I cannot see any warnings here in compiler logs, code analyzer or IDE linter. But there are warnings about the suggested version in the Analysis report: `There is no need for a @NonNull annotation because it is set as default. Only @Nullable should be used`.
Anyway, as for me it's better to see warning there then this construction.

Reply:
I certainly saw the warning in eclipse IDE. Otherwise I would not have introduced this horrendous thing

Reply:
I'm using Idea, it has pretty good code analyzer.
I think this warning can be ignored or fixed by changing signature of `org.openhab.binding.mqtt.generic.utils.FutureCollector#allOf`

Reply:
For the record, this is the warning showed by eclipse straight usage of  `xxx.stream().map(..).collect(FutureCollector.allOf())` (from Abstract `MQTTThingHandler::start`)

![image](https://user-images.githubusercontent.com/315866/205888349-7f7c008d-204d-43aa-a3fb-0a0aaf251528.png)

`Unsafe null type conversion (type annotations): The value of type '@NonNull Collector<@NonNull CompletableFuture<@Nullable Void>,@NonNull Set<@NonNull CompletableFuture<@Nullable Void>>,@NonNull CompletableFuture<@Nullable Void>>' is made accessible using the less-annotated type '@NonNull Collector<? super @NonNull CompletableFuture<@Nullable Void>,@NonNull Set<CompletableFuture<@Nullable Void>>,@NonNull CompletableFuture<@Nullable Void>>'`

However, this is not showing in SAT plugin -- only in eclipse IDE.


Sprlinking `@NonNull` into `FutureCollector::allOf` return type does not seem to make difference for eclipse:
```diff
-    public static <X> Collector<CompletableFuture<X>, Set<CompletableFuture<X>>, CompletableFuture<@Nullable Void>> allOf() {
+    public static <X> Collector<@NonNull CompletableFuture<X>, Set<@NonNull CompletableFuture<X>>, @NonNull CompletableFuture<@Nullable Void>> allOf() {
```

We still get exactly the same warning:

`Unsafe null type conversion (type annotations): The value of type '@NonNull Collector<@NonNull CompletableFuture<@Nullable Void>,@NonNull Set<@NonNull CompletableFuture<@Nullable Void>>,@NonNull CompletableFuture<@Nullable Void>>' is made accessible using the less-annotated type '@NonNull Collector<? super @NonNull CompletableFuture<@Nullable Void>,@NonNull Set<CompletableFuture<@Nullable Void>>,@NonNull CompletableFuture<@Nullable Void>>'`


It seems that the we cannot get `Set` generic type parameter to be `@NonNull`
![image](https://user-images.githubusercontent.com/315866/205891404-339076c7-8077-40d8-90d6-0e86fb6a41cc.png)

Reply:
Per your suggestion, I am opting to keep the eclipse warnings, and have cleaner code.

Reply:
See 85375d10f4

Reply:
Thanks!
It seems the only way is to remove NonNullByDefault, but we will get the SAT warning in this case.
So, let's leave it as is.